.github/workflows/ci.yml

```diff
@@ -115,8 +81,71 @@ jobs:
               -DCMAKE_COMPILE_WARNING_AS_ERROR=${{matrix.warnings-as-errors}}
             cmake --workflow --preset "${{matrix.config}}"
           fi
+
+      - name: Set capitalized config
+        run: echo "CONFIG_CAMEL=$(echo ${{ matrix.config }} | sed 's/debug/Debug/;s/release/Release/')" >> $GITHUB_ENV
+
+      - name: Upload Slang
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-${{matrix.os}}-${{matrix.platform}}-${{matrix.compiler}}-${{matrix.config}}
+          if-no-files-found: error
+          retention-days: 1
+          path: |
+            build/${{ env.CONFIG_CAMEL }}
+            !**/*.pdb
+            !**/*.ilk
+            !**/*.lib
+            !**/*.exp
+            !**/*.a
+
+      - name: Upload slang-LLVM
+        uses: actions/upload-artifact@v4
+        if: ${{ ! matrix.full-gpu-tests }}
+        with:
+          name: slang-build-${{matrix.os}}-${{matrix.platform}}-${{matrix.compiler}}-${{matrix.config}}
+          # The install directory used in the packaging step
+          path: build/dist-${{matrix.config}}/**/ZIP/slang/*
+
+  test:
+    needs: [load_matrix, build]
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.load_matrix.outputs.matrix_test) }}
+    runs-on: ${{ matrix.runs-on }}
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: '0'
+
+      - uses: ./.github/actions/common-setup
+        with:
+          os: ${{matrix.os}}
+          compiler: ${{matrix.compiler}}
+          platform: ${{matrix.platform}}
+          config: ${{matrix.config}}
+
+      - name: Set capitalized config
+        run: echo "CONFIG_CAMEL=$(echo ${{ matrix.config }} | sed 's/debug/Debug/;s/release/Release/')" >> $GITHUB_ENV
+
+      - name: Download artifacts from build job
+        uses: actions/download-artifact@v4
+        with:
+          name: test-${{matrix.os}}-${{matrix.platform}}-${{matrix.compiler}}-${{matrix.config}}
+          path: build/${{ env.CONFIG_CAMEL }}
+
+      - name: "[Debug] list files"
+        run: ls -R build
+      
       - name: Test Slang
         run: |
+          chmod a+x "$bin_dir/"*
```

Comment:
I'm very surprised that this is necessary, can we at least have a comment?

Reply:
Without this line, the executables don't have "x" permission on Linux based platforms which are all but Windows.
I think this is due to the fact that the artifact uploader doesn't preserve the file permissions.